### PR TITLE
Import push watch

### PR DESCRIPTION
**Pull request by YuvalFeldman**
Tests:
Run import and project push via cli locally
See import waiting for results
See model with validation errors present the formatted response
See model with no validation errors update that no errors were found
Models were (MNIST and LAMBDA)
Run same commands with nowait flag and ensure fire and forget not waiting for reply

image

### CODE REVIEW

File: pkg/model/api.go [hunk 1]
```diff
@@ -51,6 +53,58 @@ func ImportModel(ctx context.Context, filePath, projectId, message, modelType, b
...
+	importModelJobId := importModelData.GetImportModelJobId()
+	if waitForResults {
+		okStatus, modelId, err := waitForImportModelJob(ctx, importModelJobId)
+		if err != nil {
+			return err
+		}
+		if okStatus {
+			log.Println("Successfully imported model. ModelId: ", modelId)
+		} else {
+			log.Println("Failed to import model")
```
Comment (moshe-kabala) [inline pkg/model/api.go:65]: Consider printing the failure message - it is essential to understand what the problem is with the model

Comment (YuvalFeldman) [reply pkg/model/api.go:71]: Added

File: pkg/model/api.go [hunk 2]
```diff
@@ -51,6 +56,97 @@ func ImportModel(ctx context.Context, filePath, projectId, message, modelType, b
...
-	log.Println("Starting import model job. JobId: ", importModelData.GetImportModelJobId())
+	importModelJobId := importModelData.GetImportModelJobId()
+	if waitForResults {
+		okStatus, job, err := waitForImportModelJob(ctx, importModelJobId)
+		if err != nil {
+			return err
+		}
+		if okStatus {
+			log.Println("Successfully imported model")
+			if mappingYaml != "" {
```
Comment (moshe-kabala) [inline pkg/model/api.go:67]: can you explain why you print errors when mappingYaml is not empty?
```
// something like
someExplain := condition
```

File: pkg/model/api.go (hunk 1 above)
Comment (moshe-kabala) [reply pkg/model/api.go:71]: I mean print the error if it is possible

File: pkg/model/api.go [hunk 3]
```diff
@@ -51,6 +56,97 @@ func ImportModel(ctx context.Context, filePath, projectId, message, modelType, b
...
+				printMappingValidationErrors(ctx, job.VersionId, projectId, mappingYaml)
+			}
+		} else {
+			log.Println("Failed to import model")
```
Comment (moshe-kabala) [inline pkg/model/api.go:71]: and you need to return an error here (why usilay not return error and not printing them unless we need some special error priniting)

File: pkg/model/api.go [hunk 4]
```diff
@@ -51,6 +56,97 @@ func ImportModel(ctx context.Context, filePath, projectId, message, modelType, b
...
 }
+
+func printMappingValidationErrors(ctx context.Context, versionId string, projectId string, mapping string) {
+	getValidationErrorsParams := *tlApi.NewCodeIntegrationMappingErrorsParams(projectId, versionId, mapping)
+
+	data, _, err := api.ApiClient.GetCodeIntegrationMappingErrorsByVersionId(ctx).CodeIntegrationMappingErrorsParams(getValidationErrorsParams).Execute()
+	if err != nil {
+		log.Println("failed getting mapping validation errors:", err)
+		return
+	}
```
Comment (moshe-kabala) [inline pkg/model/api.go:135]: Consider printing some contect of the meaning of the table (e.g. We found the following errors in your current mapping file)

File: pkg/model/api.go [hunk 5]
```diff
@@ -51,6 +56,97 @@ func ImportModel(ctx context.Context, filePath, projectId, message, modelType, b
...
+	if data != nil && data.MappingErrors != nil && len(data.MappingErrors) > 0 {
+		log.Println("encountered the following validation errors while applying mapping to graph:")
+
+		table := tablewriter.NewWriter(os.Stdout)
+		table.SetHeader([]string{"Title", "Message"})
+		table.SetBorder(true)  // Set Border to false if you don't want borders
```
Comment (moshe-kabala) [inline pkg/model/api.go:141]: Remove comments

**Review by moshe-kabala — APPROVED** [review]
LGTM - looks great

File: pkg/model/api.go (hunk 2 above)
Comment (YuvalFeldman) [reply pkg/model/api.go:67]: Updated

File: pkg/model/api.go (hunk 3 above)
Comment (YuvalFeldman) [reply pkg/model/api.go:71]: Updated

File: pkg/model/api.go (hunk 4 above)
Comment (YuvalFeldman) [reply pkg/model/api.go:135]: I do add it right before the print:
encountered the following validation errors while applying mapping to graph

File: pkg/model/api.go (hunk 5 above)
Comment (YuvalFeldman) [reply pkg/model/api.go:141]: Removed

**Pull request merged**